src/Symfony/Component/Cache/LockRegistry.php

```diff
@@ -123,9 +127,17 @@ public static function compute(callable $callback, ItemInterface $item, bool &$s
                 }
                 // if we failed the race, retry locking in blocking mode to wait for the winner
                 $logger?->info('Item "{key}" is locked, waiting for it to be released', ['key' => $item->getKey()]);
-                flock($lock, \LOCK_SH);
+                if (\PHP_VERSION_ID >= 80500) {
+                    (int) flock($lock, \LOCK_SH);
```

Comment:
does the int cast work to silence the warning?
maybe that's what we could use as a better workaround everywhere, thus removing the if/else?

Reply:
> does the int cast work to silence the warning?

That's how I understand the RFC and the CI seems to agree. Using `(void)` unfortunately doesn't work on lower PHP versions.

Reply:
then let's use `(bool)` everywhere and remove the if/else?